components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -3727,9 +3734,13 @@ def get_match(name):
 
         hasObjectTypes = interfaceObject or arrayObject or dateObject or nonPlatformObject or object
         if hasObjectTypes:
-            assert interfaceObject
-            templateBody = CGList([interfaceObject], "\n")
-            conversions.append(CGIfWrapper("value.get().is_object()", templateBody))
+            assert interfaceObject or arrayObject
+            if interfaceObject:
+                templateBody = CGList([interfaceObject], "\n")
+                conversions.append(CGIfWrapper("value.get().is_object()", templateBody))
+            elif arrayObject:
+                templateBody = CGList([arrayObject], "\n")
+                conversions.append(CGIfWrapper("value.get().is_object()", templateBody))
```

Comment:
Yeah eventually other types might be added.

It seems to me that that line would be necessary in all of them though, so I'll take it out of the conditional.